packages/plugins/plugin-table/src/hooks/useTableModel.ts

```diff
@@ -2,26 +2,49 @@
 // Copyright 2024 DXOS.org
 //
 
-import { useEffect, useState } from 'react';
+import { useCallback, useEffect, useState } from 'react';
+
+import { create } from '@dxos/echo-schema';
+import { type EchoReactiveObject, getSpace } from '@dxos/react-client/echo';
 
 import { TableModel } from '../model';
 import { type TableType } from '../types';
+import { createStarterSchema, createStarterView } from '../types';
+
+export type UseTableModelParams = {
+  table: TableType;
+  objects: EchoReactiveObject<any>[];
+};
+
+export const useTableModel = ({ table, objects }: UseTableModelParams): TableModel | undefined => {
+  const space = getSpace(table);
+
+  // TODO(ZaymonFC): Not sure this belongs here. Seek feeback.
+  const onDeleteRow = useCallback(
+    (row: any) => {
+      return space?.db.remove(row);
+    },
+    [space],
+  );
+
+  // TODO(ZaymonFC): Not sure this belongs here. Seek feeback.
+  useEffect(() => {
+    if (space && !table?.schema && !table.view) {
+      table.schema = space.db.schema.addSchema(createStarterSchema());
+      table.view = createStarterView();
+      space.db.add(create(table.schema, {}));
+    }
+  }, [space, table?.schema]);
```

Comment:
@richburdon Table initialization and adding / removing things from the space feels wrong at this layer. Do you think onDeleteRow / initialization should be defined at the consumer?